packages/rum-core/src/domain/startCustomerDataTelemetry.ts

```diff
@@ -0,0 +1,143 @@
+import type { BatchFlushEvent, Context, ContextManager, Observable, Telemetry } from '@datadog/browser-core'
+import {
+  isEmptyObject,
+  includes,
+  isExperimentalFeatureEnabled,
+  performDraw,
+  ONE_SECOND,
+  addTelemetryDebug,
+  monitor,
+} from '@datadog/browser-core'
+import { RumEventType } from '../rawRumEvent.types'
+import type { RumEvent } from '../rumEvent.types'
+import type { RumConfiguration } from './configuration'
+import type { FeatureFlagContexts } from './contexts/featureFlagContext'
+import type { LifeCycle } from './lifeCycle'
+import { LifeCycleEventType } from './lifeCycle'
+
+export const MEASURES_PERIOD_DURATION = 10 * ONE_SECOND
+
+type Measure = {
+  min: number
+  max: number
+  sum: number
+}
+
+type CurrentPeriodMeasures = {
+  batchCount: number
+  batchBytesCount: Measure
+  batchMessagesCount: Measure
+  globalContextBytes: Measure
+  userContextBytes: Measure
+  featureFlagBytes: Measure
+}
+
+type CurrentBatchMeasures = {
+  globalContextBytes: Measure
+  userContextBytes: Measure
+  featureFlagBytes: Measure
+}
+
+let currentPeriodMeasures: CurrentPeriodMeasures
+let currentBatchMeasures: CurrentBatchMeasures
+
+export function startCustomerDataTelemetry(
+  configuration: RumConfiguration,
+  telemetry: Telemetry,
+  lifeCycle: LifeCycle,
+  globalContextManager: ContextManager,
+  userContextManager: ContextManager,
+  featureFlagContexts: FeatureFlagContexts,
+  batchFlushObservable: Observable<BatchFlushEvent>
+) {
+  const customerDataTelemetrySampleRate =
+    telemetry.enabled && performDraw(configuration.customerDataTelemetrySampleRate)
+  if (!customerDataTelemetrySampleRate || !isExperimentalFeatureEnabled('customer_data_telemetry')) {
+    return
+  }
+
+  initCurrentPeriodMeasures()
+  initCurrentBatchMeasures()
+
+  lifeCycle.subscribe(LifeCycleEventType.RUM_EVENT_COLLECTED, (event: RumEvent & Context) => {
+    if (!isEmptyObject(globalContextManager.getContext())) {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏`getContext` does a deepClone of the context. This is unneeded in this case, maybe use `.get()` instead? (it is deprecated, but still useful here)